### ISSUE TITLE
Projected GeoPlotters permanently modify PolyData instances

### ISSUE THREAD

**Issue by trexfeathers** [bjlittle/geovista#60]
## 🐛 Bug Report

When repeatedly plotting a `PolyData` instance with different projections, certain sequences appear to change the `PolyData`'s properties at certain points, affecting subsequent plotting attempts. I haven't been able to tell what these changes are, but the evidence is in the plots themselves.

This behaviour **is not** seen if I plot a new shallow (or deep) copy of the `PolyData`, hence the speculation that this bug is caused by `PolyData` modifications.

## How To Reproduce

<details>
<summary>Expand for code</summary>

```python
from geovista import Transform
from geovista.geoplotter import GeoPlotter
from geovista import theme
from iris import load_cube
from iris.experimental.ugrid import PARSE_UGRID_ON_LOAD
from iris.tests import get_data_path
import pyvista as pv

file_path = get_data_path(["NetCDF", "unstructured_grid", "lfric_surface_mean.nc"])

with PARSE_UGRID_ON_LOAD.context():
    my_cube = load_cube(file_path, "zh")

face_node = my_cube.mesh.face_node_connectivity
lons, lats = my_cube.mesh.node_coords
indices = face_node.indices_by_location()
my_polydata = Transform.from_unstructured(
    lons.points,
    lats.points,
    indices,
    data=my_cube.data[0],
    start_index=face_node.start_index,
    name=my_cube.name(),
)

pv.global_theme.font.color = (0.5,) * 3

globe_plotter = GeoPlotter()
globe_plotter_2 = GeoPlotter()
moll_plotter = GeoPlotter(crs="+proj=moll")
moll_plotter_90 = GeoPlotter(crs="+proj=moll +lon_0=90")

for ix, plotter in enumerate([globe_plotter, moll_plotter, moll_plotter_90, globe_plotter_2]):
    plotter.add_mesh(my_polydata, show_edges=True)
    print(my_polydata.cell_arrays)
    plotter.camera_position = "xy"
    plotter.show(screenshot=f"{ix}.png")

```

</details>

<img src="https://user-images.githubusercontent.com/40734014/162478415-1a2fada3-38f5-416b-9673-26d98f556ab4.png" width="200"/> <img src="https://user-images.githubusercontent.com/40734014/162478419-8b74b831-9560-441c-a628-93f24b479340.png" width="200"/> <img src="https://user-images.githubusercontent.com/40734014/162478422-fb9d6f89-d562-4081-b1d1-94a771473ec6.png" width="200"/> <img src="https://user-images.githubusercontent.com/40734014/162478424-37f37de7-c541-4dc4-8910-37260f1b9a2d.png" width="200"/>

## Expected behaviour

<img src="https://user-images.githubusercontent.com/40734014/162478415-1a2fada3-38f5-416b-9673-26d98f556ab4.png" width="200"/> <img src="https://user-images.githubusercontent.com/40734014/162478419-8b74b831-9560-441c-a628-93f24b479340.png" width="200"/> <img src="https://user-images.githubusercontent.com/40734014/162479419-adb3ae70-44b5-41a0-9852-00e1e05375bb.png" width="200"/> <img src="https://user-images.githubusercontent.com/40734014/162478415-1a2fada3-38f5-416b-9673-26d98f556ab4.png" width="200"/>

## Environment 

* OS & Version - Linux RHEL7
* GeoVista Version - `0.1a1.dev323+dirty`

**Comment by bjlittle** [comment]
Thanks @trexfeathers 

This is just the initial `GeoPlotter` functionality which changes the `PolyData` in place.

I'm going to abstract this out to make projections of `PolyData` independent of the`GeoPlotter` and give controls to transform inplace or not (non-inplace can be expensive) 👍

**Comment by trexfeathers** [comment]
@bjlittle is this fixed yet?

**Comment by bjlittle** [comment]
@trexfeathers This issue is now resolved.

It was a simple case of better mesh active scalars management.

The resultant plots are now as expected i.e.,

### plot: 0.png
`GeoPlotter()`
![0](https://user-images.githubusercontent.com/2051656/215813472-ff9dc460-0170-494b-87ab-edf69ddbe45f.png)

### plot: 1.png
GeoPlotter(crs="+proj=moll")
![1](https://user-images.githubusercontent.com/2051656/215813835-c8f97cbe-4dfb-4fab-92a5-0bb53837c2d3.png)

### plot: 2.png
GeoPlotter(crs="+proj=moll +lon_0=90")
![2](https://user-images.githubusercontent.com/2051656/215813901-747ded87-5850-435a-9d30-a02f380298d1.png)

### plot: 3.png
`GeoPlotter()`
![3](https://user-images.githubusercontent.com/2051656/215813945-f44671ab-9df3-4c1e-ba7d-f2dd4cdca892.png)

**Comment by bjlittle** [comment]
@trexfeathers I'm going to close this issue. Thanks for raising it, and apologies for the super tardy response.

The fix is available from `0.1a1.dev563`.

Please re-run your example script to confirm that you're getting the same behaviour and satisfied that this issue has been addressed. Otherwise, re-open this issue and I'll investigate further.

Cheers!

**Comment by trexfeathers** [comment]
I can confirm that the issue is fixed. Not that I was only able to do multiple plots using VTK version `9.2.2` (we suspect something is wrong with version `9.2.5`).
